src/app/static/src/tests/components/plots/barchart/filterSelect.test.ts

```diff
@@ -3,17 +3,19 @@ import {createLocalVue, shallowMount} from "@vue/test-utils";
 
 const localVue = createLocalVue();
 
-const propsData = {
-    id: "test",
-    options: [{id: "fo1", label: "option 1"}, {id: "fo2", label: "option 2"}],
-    isXAxis: true,
-    isDisaggregateBy: false,
-    value: [{id: "fo2", label: "option 2"}],
-    label: "Test Filter"
+const defaultProps = () => {
+    return {
+        id: "test",
+        options: [{id: "fo1", label: "option 1"}, {id: "fo2", label: "option 2"}],
+        isXAxis: true,
+        isDisaggregateBy: false,
+        value: [{id: "fo2", label: "option 2"}],
+        label: "Test Filter"
+    }
 };
 
 const getWrapper = (props: any = {}) => {
-    return shallowMount(FilterSelect, {propsData: {...propsData, ...props}, localVue});
+    return shallowMount(FilterSelect, {propsData: {...defaultProps(), ...props}, localVue});
```

Comment:
This was a bit weird - propsData seemed to get mutated somewhere in the course of these tests running, even though they were only being used with ...   Changing to a function seemed to sort it, but it's bizarre...

Reply:
Was it the `value` or `options` properties that were mutated? spread operator will only give a shallow clone, could be why.

Reply:
Ah, yeah I think you're right. It was `value` that was causing the problems - I guess the clone just got a pointer to the same array.